### ISSUE TITLE
Better support for true serverless functions

### ISSUE THREAD

**Issue by wilzbach** [microservices/microservice.guide#77]
One cool part of AWS Lambda or OpenFaas is that you directly get your Python/NodeJs function that gets called and you just need to return from this function. No webserver setup required.
Currently, it requires more effort for a user to create a small microservice which is absolutely okay, because OMG needs to be flexible.

At the moment OMG has two simple interfaces:
- HTTP (requires starting a web server)
- Console (requires parsing arguments from stdout)

The `console` interface is already very close and reduces the "boilerplate" to 3 lines:

```py
def main(arg1, arg2):
    print(arg1, arg2)


if __name__ == '__main__':
    from sys import argv
    main(*argv[1:])
```

However, it has three big disadvantages:
- no auto-serialization (it's just strings)
- it's still code that a high-level user shouldn't care about
- starting a process in an interpreted language like Python/NodeJS does come with an overhead

Now, of course the serialization bit could be handled with two more lines:

```
def main(arg1, arg2):
    print(arg1, arg2)


if __name__ == '__main__':
    from sys import argv
    from json import loads
    main(*map(loads, argv[1:]))
```

But it's still not perfect, because the number of characters that can be processed via the CLI has a limit (see e.g. https://serverfault.com/questions/163371/linux-command-line-character-limit, https://www.in-ulm.de/~mascheck/various/argmax/).

tl;dr: `console` should only be used for testing.

So this brings us back to the problem: do we want to abstract the underlying web server away from the user? They probably don't care in many cases and we could tweak it for the Asyncy Cloud.

Here's one proposal:

```
FROM        omg/python:3.7-alpine

ADD         requirements.txt /app
RUN         pip install -r /app/requirements.txt
ADD         app.py /app
```

(entrypoint is predefined)

The rough idea here would be to:
- have a template image 
- use `/app/app.py` as convention to detect functions
- create a route for every function found
- allow to generate/update the `actions` in `microservice.yml` based on the generated routes

As this requires some implicit conventions, I'm no longer sure that it's a good idea, but I wanted to put it out there nevertheless.
Moreover, we could also do only a partial solution where we only add tools to help with updating the `microservice.yml` based on the exposed functions.
